R/layers.R

```diff
@@ -177,24 +177,125 @@ popupOptions = function(
   )
 }
 
-#' @param icon the icon for markers; if you want to create a new icon using
-#'   JavaScript, please remember to use \code{\link[htmlwidgets]{JS}()} on the
-#'   JavaScript string; see \url{http://leafletjs.com/reference.html#icon}
+#' @param icon the icon(s) for markers; you can either use literal JavaScript
+#'   code (e.g. \code{\link[htmlwidgets]{JS}('L.icon({iconUrl: '?', size: [?,
+#'   ?]})')} or an R list to create an icon (e.g. \code{list(iconUrl = '?', size
+#'   = c(x, y))}), or use \code{\link{iconList}()} to create a list of icons;
+#'   note when you use an R list that contains images as local files (tested by
+#'   \code{file.exists()}), these local image files will be base64 encoded into
+#'   the HTML page so the icon images will still be available even when you
+#'   publish the map elsewhere (use \code{iconList(..., embed = FALSE)} if you
+#'   do not want the images to be encoded and embedded)
+#' @param iconGroup when \code{icon} contains a list of different icons, this
+#'   argument is a numeric vector of the same length as \code{lat} and
+#'   \code{lng}, where \code{iconGroup[i]} is the index of the icon in the
+#'   \code{icon} list for \code{lat[i]} and \code{lng[i]}; the main purpose of
+#'   this argument is to reduce the size of the icon data when the icon images
+#'   are base64 encoded: it is not efficient to encode the same image multiple
+#'   times when the image is used multiple times on the map, and we can encode
```

Comment:
I think about this in a rather different way - I think you are creating a scale that maps between a categorical variable and a list of shapes. I think it would be better to call this argument `icon` and allow it to take three possible values:

* an icon object (in which case the same icon is used for everything)
* an integer vector (which indexes into the icon list)
* a character vector (which indexes into the icon list by name)